documentation/src/docs/asciidoc/user-guide/writing-tests.adoc

```diff
@@ -1830,6 +1830,10 @@ implementations.
   refer to the Javadoc for `DiscoverySelectors.selectMethod(String)` for the supported
   formats for a FQMN.
 
+`ClassSource` ::
+  If the `URI` contains the `class` scheme and the fully qualified class name --
+  for example, `class:/org.junit.Foo?line=42`.
```

Comment:
❓ Is there a technical reason why we require a leading slash here? I think `class:org.junit.Foo?line=42` would be more consistent with the `method` scheme.

Reply:
It's the distinction between opaque and hierarchical URIs - paths that do not start with a slash are opaque, and are more work to extract the query from, e.g.:
```
Stream.of(
    "scheme:path?query#fragment",
    "scheme:/path?query#fragment" )
    .forEach( s -> {
	    try {
		    URI uri = new URI( s );
		    System.out.println( String.format(
		        "uri '%s' scheme='%s' ssp='%s' path='%s' query='%s' fragment='%s'",
		        uri,
		        uri.getScheme(),
		        uri.getSchemeSpecificPart(),
		        uri.getPath(),
		        uri.getQuery(),
		        uri.getFragment() ) );
	    }
	    catch ( Exception e ) {
		    e.printStackTrace();
	    }
    } );
```
produces
```
uri 'scheme:path?query#fragment' scheme='scheme' ssp='path?query' path='null' query='null' fragment='fragment'
uri 'scheme:/path?query#fragment' scheme='scheme' ssp='/path?query' path='/path' query='query' fragment='fragment'
```

I do agree that it's worth the effort to get rid of that leading slash though, I'll make the change